Lint session fails following tests-3.x session runs

<!-- Thank you for filing a bug! Please feel free to answer as much or as little of this template as you can. -->

**Describe the bug**
In the nox repository, when running the `lint` session after having run one of the `tests-3.x` sessions, flake8 fails on analysis of the test generated virtual environments in `tests/resources/.nox/`:

```
nox > flake8 nox tests noxfile.py setup.py
... (lines skipped)
tests/resources/.nox/test-arg-datetime-datetime-1980-1-1-0-0/lib/python3.9/site-packages/wheel/bdist_wheel.py:140:72: W504 line break after binary operator
nox > Command flake8 nox tests noxfile.py setup.py failed with exit code 1
nox > Session lint failed.

```

**How to reproduce**

Run `nox -s tests-3.8 lint`

**Expected behavior**
<!-- What should have happened? -->
The `lint` session passes without errors.

**Notes**

- This can be fixed by adding `exclude = .nox` in the `.flake8` file.
- This is not detected in CI since the `lint` job runs in isolation from the test jobs.

--- a/tests/resources/noxfile_normalization.py
+++ b/tests/resources/noxfile_normalization.py
@@ -7,7 +7,7 @@ class Foo:
     pass
 
 
-@nox.session
+@nox.session(venv_backend="none")
 @nox.parametrize(
     "arg",
     ["Jane", "Joe's", '"hello world"', datetime.datetime(1980, 1, 1), [42], Foo()],

